src/bootupd.rs

```diff
@@ -27,6 +26,7 @@ use std::path::Path;
 mod cli;
 pub use cli::MultiCall;
 mod component;
+mod daemon;
```

Comment:
This is great, I had started on more clearly splitting the daemon/client but shelved it.

Reply:
Thanks for the feedback, I wasn't sure about the code layout so I did it here opportunistically in a separate commit (for easy dropping). I'm taking a mental note, if I happen to touch more of client logic I'll go for a similar setup.

Reply:
Mostly same setup, a bit of spacemacs too. But to avoid brain meltdown I try not to have too many tabs in a row, and prefer split windows. Although there are other breadcrumb/hints, I can see the `mod.rs` overcrowding you mention, indeed that PR would help.